src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -1030,15 +1032,31 @@ public IEnumerable<IExpandInfo> GetExpands()
 
             return expands;
         }
+        
+        public DType ToRecord()
+        {
+            var fError = false;
+            var type = ToRecord(ref fError);
+            
+            if (fError) 
+            {
+                Contracts.Assert(false, "Bad source kind for ToRecord");
+            }
+
+            return type;
+        }
 
         public DType ToRecord(ref bool fError)
         {
             AssertValid();
 
             switch (Kind)
             {
+                case DKind.LazyRecord:
                 case DKind.Record:
                     return this;
+                case DKind.LazyTable:
```

Comment:
A LazyTable/LazyRecord that are backed by the same schema use the same LazyTypeProvider, so switching between them is just constructing a new DType with the appropriate DKind.